### PR TITLE
Use the core feed

### PR DESCRIPTION
**Pull request by RadoslavGeorgiev**
With https://github.com/woocommerce/woocommerce/pull/62313 in place, this PR removes everything that is already in core.

### Testing Instructions

Test the CLI command and automatic pushing, like it was done in other PRs. Just make sure that you have a fresh `trunk` of WooCommerce.

For tests, you need to modify the local environment to point to Woo instead of downloading it. Here is how I did it:

```diff
diff --git a/.wp-env.json b/.wp-env.json
index 2ac8756..31d49b9 100644
--- a/.wp-env.json
+++ b/.wp-env.json
@@ -1,7 +1,7 @@
 {
 	"phpVersion": "8.1",
 	"plugins": [
-		"https://downloads.wordpress.org/plugin/woocommerce.latest-stable.zip",
+		"/Users/rado/Projects/woocommerce/plugins/woocommerce",
 		"."
 	],
 	"mappings": {

```

## Summary by CodeRabbit

* **Refactor**
  * Migrate product feed to use internal feed types and central container wiring; simplify activation flow and add runtime guards.
  * Add safety checks to avoid errors when the scheduler or logger are unavailable.

* **Removals**
  * Remove POS Catalog integration, async generation, product feed walker, file-backed JSON feed, several feed/mapping/validation contracts, and utility helpers.

* **Tests**
  * Remove or update numerous unit tests associated with the removed components.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/Integrations/OpenAi/OpenAiIntegration.php [hunk 1]
```diff
@@ -9,15 +9,15 @@
 
 namespace Automattic\WooCommerce\ProductFeedForOpenAI\Integrations\OpenAi;
 
+use Automattic\WooCommerce\Internal\ProductFeed\Feed\FeedInterface;
+use Automattic\WooCommerce\Internal\ProductFeed\Feed\FeedValidatorInterface;
+use Automattic\WooCommerce\Internal\ProductFeed\Feed\ProductMapperInterface;
+use Automattic\WooCommerce\Internal\ProductFeed\Integrations\IntegrationInterface;
+use Automattic\WooCommerce\Internal\ProductFeed\Storage\JsonFileFeed;
```
[the remark this answers is not in the document]
Comment (ebinnion) [reply src/Integrations/OpenAi/OpenAiIntegration.php:16]: We should probably do this. Also, of note, I don't think that there's any explicit check for if WooCommerce is active. I get the following error when Woo is deactivated on my site.

```
[16-Dec-2025 01:54:28 UTC] PHP Fatal error:  Uncaught Error: Call to undefined function Automattic\WooCommerce\ProductFeedForOpenAI\Core\DependencyManagement\wc_get_logger() in /var/www/html/wp-content/plugins/woocommerce-product-feed-for-openai/src/Core/DependencyManagement/Container.php:36
Stack trace:
#0 /var/www/html/wp-content/plugins/woocommerce-product-feed-for-openai/src/Core/Plugin.php(49): Automattic\WooCommerce\ProductFeedForOpenAI\Core\DependencyManagement\Container->__construct()
#1 /var/www/html/wp-content/plugins/woocommerce-product-feed-for-openai/src/Core/Plugin.php(40): Automattic\WooCommerce\ProductFeedForOpenAI\Core\Plugin->__construct()
#2 /var/www/html/wp-includes/class-wp-hook.php(341): Automattic\WooCommerce\ProductFeedForOpenAI\Core\Plugin::get_instance('')
#3 /var/www/html/wp-includes/class-wp-hook.php(365): WP_Hook->apply_filters(NULL, Array)
#4 /var/www/html/wp-includes/plugin.php(522): WP_Hook->do_action(Array)
#5 /var/www/html/wp-settings.php(593): do_action('plugins_loaded')
#6 /var/www/html/wp-config.php(145): require_once('/var/www/html/w...')
#7 /var/www/html/wp-load.php(50): require_once('/var/www/html/w...')
#8 /var/www/html/wp-admin/admin-ajax.php(22): require_once('/var/www/html/w...')
#9 {main}
  thrown in /var/www/html/wp-content/plugins/woocommerce-product-feed-for-openai/src/Core/DependencyManagement/Container.php on line 36
```

**Review by ebinnion — APPROVED** [review]
I validated that the files removed here are added in core and did some smoke testing of the CLI. So, we should be good. I left a non-blocking comment about adding a more explicit check for whether Woo is enabled.

**Comment by RadoslavGeorgiev** [comment]
@ebinnion I'm just taking a look now. I think you might have forgotten to post the comment regarding explicit checks?

**Comment by ebinnion** [comment]
@RadoslavGeorgiev  - Yea. I guess so. The issue I found is that if this extension is enabled while WooCommerce is disabled, there is a fatal error. It's not related to this PR though.

File: src/Core/Plugin.php [hunk 2]
```diff
@@ -62,22 +54,20 @@ private function __construct() {
...
+			try {
+				$product_feed = wc_get_container()->get( ProductFeed::class );
+				$product_feed->register_integration( $open_ai_integration );
+			} catch ( \Throwable $e ) {
+				// ProductFeed service not available in this WooCommerce version.
+				return;
+			}
+
+			$open_ai_integration->register_hooks();
 		}
```
[the remark this answers is not in the document]
Comment (RadoslavGeorgiev) [reply src/Core/Plugin.php]: Is this still an issue?

File: src/Core/Plugin.php [hunk 3]
```diff
@@ -107,18 +97,14 @@ public function activate(): void {
 			);
 		}
 
-		foreach ( $this->integration_registry->get_integrations() as $integration ) {
-			$integration->activate();
-		}
+		$this->container->get( OpenAiIntegration::class )->activate();
 	}
```
[the remark this answers is not in the document]
Comment (RadoslavGeorgiev) [reply src/Core/Plugin.php:107]: Is this still an issue?

File: src/Integrations/OpenAi/OpenAiIntegration.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (RadoslavGeorgiev) [reply src/Integrations/OpenAi/OpenAiIntegration.php:16]: This should be fixed. CodeRabbit, is this still an issue?

[the remark this answers is not in the document]
Comment (RadoslavGeorgiev) [reply src/Integrations/OpenAi/OpenAiIntegration.php:16]: @coderabbitai I'm asking you. I think the issue is fixed, I'm looking for verification.
